File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool cmp(int a, int b){
    return a > b;
}

int solution(int k, vector<int> tangerine) {
    int answer = 0;
    
    vector<int> v(10000001, 0);
    
    for(int i=0; i<tangerine.size(); i++){
        v[tangerine[i]]++;
    }
    
    sort(v.begin(), v.end(), cmp);
    
    int i=0;
    for (int i = 0; i < v.size() && k > 0; i++) {
        k -= v[i];
        answer++;
    }
    
    
    return answer;
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.java
import java.util.*;

class Solution {
    
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        
        HashMap<Integer, Integer> map = new HashMap<>();
        
        for(int i=0; i<tangerine.length; i++){
            map.put(tangerine[i], map.getOrDefault(tangerine[i], 0)+1);
        }
        
        ArrayList<Integer> list = new ArrayList<>();
        for(int i : map.keySet()){
            list.add(map.get(i));
        }
        
        Collections.sort(list, Collections.reverseOrder());
        
        for(int i=0; i<list.size(); i++){
            if(k<=0) break;
            
            k -= list.get(i);
            answer++;
        }
        
        
        return answer;
    }
}